src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -323,6 +332,50 @@ void CollectError(Type extensionType, Version minimumVersion, ExtensionStartupTy
             }
         }
 
+        private bool IsValidBindingMatch(ExtensionReference extension, HashSet<string> bindingsSet)
+        {
+            if (extension.Bindings.Count == 0)
+            {
+                return true;
+            }
+
+            // Application Insights uses a special binding value to ensure it's installed based on its App Setting
+            if (extension.Bindings.Any(b => b.Equals("_")))
+            {
+                return IsApplicationInsightsSettingPresent();
+            }
+
+            return extension.Bindings.Intersect(bindingsSet, StringComparer.OrdinalIgnoreCase).Any();
+        }
+
+        private void ValidateApplicationInsightsConfig(List<Type> startupTypes)
+        {
+            if (_isSelfHost)
+            {
+                _logger.LogWarning("In order to use Application Insights in Azure Functions V4 and above, please install the Application Insights Extension.");
+            }
+            else
+            {
+                bool applicationInsightsInstalled = startupTypes.Exists(t => t.Name.Equals("ApplicationInsightsWebJobsStartup"));
+                if (applicationInsightsInstalled && !IsApplicationInsightsSettingPresent())
+                {
+                    throw new HostInitializationException($"The Application Insights Extension is installed but is not properly configured. " +
+                                                          $"Please define the \"{EnvironmentSettingNames.AppInsightsConnectionString}\" app setting.");
+                }
+                else if (!applicationInsightsInstalled && IsApplicationInsightsSettingPresent())
+                {
+                    throw new HostInitializationException($"{EnvironmentSettingNames.AppInsightsConnectionString} or {EnvironmentSettingNames.AppInsightsInstrumentationKey} " +
+                                                          $"is defined but the Application Insights Extension is not installed. Please install the Application Insights Extension.");
```

Comment:
Would this break all extension bundle-configured apps if this change is released and used before extension bundles has been released with the App Insights extension?

Reply:
Yes -- could we have another check to see if this is using bundles and only throw if it is not?